src/prof.c

```diff
@@ -1790,6 +1790,7 @@ prof_leakcheck(const prof_cnt_t *cnt_all, size_t leak_ngctx,
 		malloc_printf(
 		    "<jemalloc>: Run jeprof on \"%s\" for leak detail\n",
 		    filename);
+		_exit(1);    // we don't want to call atexit() routines!
```

Comment:
This function is invoked through `atexit` already, via `prof_fdump`: https://github.com/jemalloc/jemalloc/blob/dev/src/prof.c#L646

Since this is changing the behavior / not backward compatible, we will need to guard it with a runtime option, e.g. maybe `opt.prof_leak_error`?

The other thing is, you need to work on the dev branch, and submit PR for dev only. The master branch is only for releases. Closing but feel free to ask anything.

Reply:
nice suggestion, thank u

Reply:
In case you want an example of adding runtime options, here is one: https://github.com/jemalloc/jemalloc/pull/2027/files